### PR TITLE
Add compat annotation for NaN handling in (l|r)mul!

### PR DESCRIPTION
**Pull request by tkf**
As discussed in JuliaLang/LinearAlgebra.jl#560, the behavior of `lmul!` and `rmul!` was changed between 1.0 and 1.1.  I think issue JuliaLang/LinearAlgebra.jl#560 should be treated as an API documentation bug (and I think it would be nice if similar issues are closed by PRs addressing the issue).  This PR adds docstring to elaborate the API of those functions and adds compat annotation for the incompatible change.

### CODE REVIEW

**Comment by tkf** [comment]
I just noticed #29726 has more discussion and it was considered actually a bug fix https://github.com/JuliaLang/julia/pull/29726#issuecomment-442347850

Nevertheless, it's good to have docs and tests.

File: stdlib/LinearAlgebra/src/generic.jl [hunk 1]
```diff
@@ -31,7 +31,15 @@ mul!(C::AbstractArray, X::AbstractArray, s::Number) = generic_mul!(C, s, X)
...
 
-Scale an array `A` by a scalar `b` overwriting `A` in-place.
+Scale an array `A` by a scalar `b` overwriting `A` in-place.  Use
+[`lmul!`](@ref) to multiply scalar from left.  The scaling operation
+respects the semantics of the multiplication [`*`](@ref) between an
+element of `A` and `b`.  In particular, `NaN` entries in `A` are not
+modified (unless the type of `b` defines a non-standard behavior).
+
+!!! compat "Julia 1.1"
+    Prior to Julia 1.1, how `NaN` entries in `A` were treated was
```
Comment (andreasnoack) [inline stdlib/LinearAlgebra/src/generic.jl]: Maybe rephrase to something like `Prior to Julia 1.1, `NaN` entries in `A` were treated inconsistently.`

Comment (tkf) [reply stdlib/LinearAlgebra/src/generic.jl]: Thanks, that's much smoother.  I changed the docstring.  I also mention `Inf` since that's also changed.

**Review by andreasnoack — APPROVED** [review]
